File: app/src/main/java/com/kode/utair/presentation/mvp/views/WeatherForecastView.java
package com.kode.utair.presentation.mvp.views;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;


@StateStrategyType(AddToEndSingleStrategy.class)
public interface WeatherForecastView extends MvpView {

    void showCitiesNames(String departCityName, String arriveCityName);
    void setTabSelected(int currentTabPosition);
    void openForecastPage(int position);
    void showForecastForCities(String departCity, String arriveCity);

}

File: app/src/main/java/com/utair/presentation/mvp/weatherforecast/WeatherForecastView.kt
package com.utair.presentation.mvp.weatherforecast

import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

@StateStrategyType(AddToEndSingleStrategy::class)
interface WeatherForecastView : MvpView {

    fun showCitiesNames(departCityName: String, arriveCityName: String)

    fun setTabSelected(currentTabPosition: Int)

    fun openForecastPage(position: Int)

    fun showForecastForCities(departCity: String, arriveCity: String)

}